### PR TITLE
very simple initial TriggerPrimitive decoder

### PR DESCRIPTION
**Pull request by wesketchum**
Don't know why, but this got merged to develop but somehow missed the production/v4 branch? Opening a PR to production/v4 directly.

### CODE REVIEW

File: scripts/tpdecoder.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import h5py
+
+import daqdataformats
+import trgdataformats
+
+import click
+import time
+import numpy as np
+
+from rawdatautils.unpack.dataclasses import *
```
Comment (aeoranday) [inline scripts/tpdecoder.py]: Why import all if you only need `TriggerPrimitiveData`?

File: scripts/tpdecoder.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        records_to_process = []
+        if nrecords==-1:
+            records_to_process = records[nskip:]
+        else:
+            records_to_process = records[nskip:nrecords]
+        print(f'Will process {len(records_to_process)} of {len(records)} records.')
+
+        
+        with h5py.File(h5_file.get_file_name(), 'r') as f:
+            record_type = f.attrs["record_type"]
```
Comment (aeoranday) [inline scripts/tpdecoder.py:46]: Was this intended for something?

File: scripts/tpdecoder.py
Comment (aeoranday) [inline scripts/tpdecoder.py:1]: The mode isn't set to 755 like the other scripts, and this is also the case for the version in `develop`. Should this be changed?

**Review by aeoranday — APPROVED** [review]
LGTM. I ran this and found no issues.

**Review by bieryAtFnal — CHANGES_REQUESTED** [review]
I was able to run the script successfully on both raw data and TP stream files.  

Like Alex, I wondered why this file doesn't have execute permissions...

File: scripts/tpdecoder.py (hunk 2 above)
Comment (wesketchum) [reply scripts/tpdecoder.py:46]: It's not used, but was intended to be able to extract if it's a trigger record or timeslice record. I've left it in, but adding a print statement to inform.

File: scripts/tpdecoder.py
Comment (wesketchum) [reply scripts/tpdecoder.py:1]: Fixed.

**Review by aeoranday — APPROVED** [review]
Ran over a few raw and tp files and found that it works for both. LGTM.

**Review by bieryAtFnal — APPROVED** [review]
LGTM

**Pull request merged**